numpy/testing/dec.py

```diff
@@ -0,0 +1,2 @@
+import numpy as np
+from numpy.testing._private.decorators import *
```

Comment:
What's the motivation for this change?

Reply:
Note that the decorators are only retained for folks using the old nose framework and will be removed as some point, nose doesn't work with recent Python without modification. Most distros no longer provide nose, but rather nose2.

Reply:
The purpose of the changes is to suppress warnings mentioned above (`no module named 'dec' / 'char'`). `numpy.testing.dec` and `numpy.char` are aliases, and the names of modules are different (`numpy.testing._private.decorators` and `numpy.core.defchararray`). I think this is the cause of warnings.

Reply:
Are these warnings new in sphinx3?

Reply:
Yes. Sphinx 2 doesn't emit these warnings.

Reply:
Do any of our docs reference `np.testing.dec`?

Reply:
Ah yes: https://numpy.org/devdocs/reference/routines.testing.html#decorators

Reply:
I worked around this, and the `numpy.char` import error, in a slightly different way which *might* be preferable but I leave it to your judgment:

The import errors can be suppressed by "caching" the modules that the import machinery is looking for. After `import numpy` in `doc/source/conf.py`, I inserted the following two lines:

```python
sys.modules['numpy.char'] = numpy.char
sys.modules['numpy.testing.dec'] = numpy.testing.dec
```

I think this *might* be preferable as:

* It only modifies the sphinx environment, not the numpy library itself
* It doesn't involve adding another `from X import *`
* If those aliases-of-modules should change into "real" modules in the library itself, it won't affect the work-around

Reply:
I've already put way scarier hacks in `conf.py` - in the interest of not having to decide about subtle and largely irrelevant API changes, I'd be in favor of doing it the way you suggest @waveform80, and considering the approach we tried here as a follow up.

Reply:
I also think it's better. Fixed.